tests/profiling_v2/collector/test_threading.py

```diff
@@ -88,94 +91,67 @@ def test_repr(
     test_collector._test_repr(collector_class, expected_repr)
 
 
-@pytest.mark.parametrize(
-    "lock_class,collector_class",
-    [
-        (threading.Lock, ThreadingLockCollector),
-        (threading.RLock, ThreadingRLockCollector),
-    ],
-)
```

Comment:
I removed it because the patching mechanism is now exactly the same for both Lock and RLock:
1. Store the original lock factory in _original_lock
2. Replace it with a _LockAllocatorWrapper that creates _ProfiledLock instances
3. Restore the original on unpatch

We could keep it parametrized, but it would literally check the same codepath twice. I introduced param-ion recently with #14823 , but it's not necessary here anymore.